### PR TITLE
Enable nullable reference types

### PR DESCRIPTION
**Pull request by sharwell**

### CODE REVIEW

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs [hunk 1]
```diff
@@ -588,7 +588,6 @@ private Solution GetSolution((string filename, SourceText content)[] sources, (s
         /// strings.</returns>
         protected Project CreateProject((string filename, SourceText content)[] sources, (string filename, SourceText content)[] additionalFiles, ProjectState[] additionalProjects, MetadataReference[] additionalMetadataReferences, string language)
         {
-            language = language ?? language;
```
Comment (jmarolf) [inline src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs]: Why the behavior change?

Comment (jmarolf) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs]: actually I have no idea what this was trying to do before

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/TestFileMarkupParser.cs [hunk 2]
```diff
@@ -204,6 +204,10 @@ public static void GetPositionAndSpans(string input, out int? cursorPositionOpt,
         public static void GetPositionAndSpans(string input, out string output, out int cursorPosition, out IDictionary<string, IList<TextSpan>> spans)
         {
             GetPositionAndSpans(input, out output, out int? cursorPositionOpt, out spans);
+            if (cursorPositionOpt is null)
+            {
+                throw new InvalidOperationException();
```
Comment (jmarolf) [inline src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/TestFileMarkupParser.cs]: Can we provide a better error message here to the user?

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs (hunk 1 above)
Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/AnalyzerTest`1.cs]: > actually I have no idea what this was trying to do before

Exactly 😂

File: src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/TestFileMarkupParser.cs (hunk 2 above)
Comment (sharwell) [reply src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/TestFileMarkupParser.cs]: Done!

**Pull request merged**